<launch>
    <!-- Launches the AR.Drone driver -->
        <node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" output="screen" clear_params="true"></node>

    <!-- Joy Node -->
        <node name="joy_node" pkg="joy" type="joy_node" output="screen" respawn="true" clear_params="true">
        <param name="dev" type="string" value="/dev/input/js1"/>
    </node>

    <!-- Launches the joystick controller -->
    <node name="aldrone_teleop" pkg="aldrone_utils" type="aldrone_teleop" output="screen" required="true">

        <!-- Configures the joystick button mapping -->
        <param name="btn_emergency"        type="int"     value="0" />
        <param name="btn_land"             type="int"     value="1" />
        <param name="btn_takeoff"          type="int"     value="8" />

        <!-- Configures the joystick axes mapping -->
        <param name="axis_x"               type="int"     value="0" />
        <param name="axis_y"              type="int"     value="1" />
        <param name="axis_z"                type="int"     value="3" />
        <param name="axis_yaw"                  type="int"     value="2" />
        
        <!-- Configures the joystick scales -->
        <param name="scale_x"              type="double"     value="1.0" />
        <param name="scale_y"             type="double"     value="1.0" />
        <param name="scale_z"               type="double"     value="1.0" />
        <param name="scale_yaw"                 type="double"     value="1.0" />

    </node>
</launch>
